nameko/__init__.py

```diff
@@ -1,8 +1,71 @@
 from copy import deepcopy
+from six import wraps
+from six.moves import UserDict
 
 
-config = {}
+class Config(UserDict):
+
+    def setup(self, config):
+        self.data.clear()
+        self.data.update(config)
+
+    def patch(self, context_config, clear=False):
+        """
+        A context manager and decorator for updating config just for the given context
+        or decorated function execution.
+
+        Intended to be used mainly in tests:
+
+            @pytest.yield_fixture
+            def memory_rabbit_config():
+                with nameko.config.patch({"AMQP_URI": "memory://"}):
+                    yield
+
+            @nameko.config.patch({"AMQP_URI": "memory://"})
+            def test_spam():
+                assert nameko.config["AMQP_URI"] == "memory://"
+
+        If `clear` is set to True then the config is completely replaced with the given
+        `context_config` otherwise it's only updated.
+
+        """
+
+        config = self.data
```

Comment:
Might be clearer to call this `data`. I was briefly confused and thought the implementation below was writing to the _global_ config var.

Reply:
:+1: